From 88481baa83b5eedc4506ea2fb76f3186443fbb2e Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Fri, 18 Jun 2010 16:11:37 -0300
Subject: [PATCH 3/8] smbios: allow vendor/manufacturer/version/product names be set on config.h

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1276877502-7483-3-git-send-email-ehabkost@redhat.com>
Patchwork-id: 10001
O-Subject: [RHEL6 seabios PATCH 2/7] smbios: allow
	vendor/manufacturer/version/product names be set on config.h
Bugzilla: 593317
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

Bugzilla: 593317
Upstream status: will be submitted today

Instead of using the same string for every field, let those fields to be set
individually on config.h

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 src/config.h |    8 ++++++++
 src/smbios.c |   16 ++++++++--------
 2 files changed, 16 insertions(+), 8 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 src/config.h |    8 ++++++++
 src/smbios.c |   12 ++++++------
 2 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/src/config.h b/src/config.h
index 1065d26..676e474 100644
--- a/src/config.h
+++ b/src/config.h
@@ -12,6 +12,14 @@
 #define CONFIG_APPNAME6 "BOCHS "
 #define CONFIG_APPNAME4 "BXPC"
 
+/* SMBIOS fields */
+#define CONFIG_SMBIOS_VENDOR CONFIG_APPNAME
+#define CONFIG_SMBIOS_VERSION CONFIG_APPNAME
+#define CONFIG_SMBIOS_MANUFACTURER CONFIG_APPNAME
+#define CONFIG_SMBIOS_PROD_NAME CONFIG_APPNAME
+#define CONFIG_SMBIOS_TYPE3_NAME CONFIG_APPNAME
+#define CONFIG_SMBIOS_CPU_NAME CONFIG_APPNAME
+
 // Configure as a coreboot payload.
 #define CONFIG_COREBOOT 0
 
diff --git a/src/smbios.c b/src/smbios.c
index 8df0f2d..7416e6f 100644
--- a/src/smbios.c
+++ b/src/smbios.c
@@ -97,8 +97,8 @@ smbios_init_type_0(void *start)
     p->header.length = sizeof(struct smbios_type_0);
     p->header.handle = 0;
 
-    load_str_field_with_default(0, vendor_str, CONFIG_APPNAME);
-    load_str_field_with_default(0, bios_version_str, CONFIG_APPNAME);
+    load_str_field_with_default(0, vendor_str, CONFIG_SMBIOS_VENDOR);
+    load_str_field_with_default(0, bios_version_str, CONFIG_SMBIOS_VERSION);
 
     p->bios_starting_address_segment = 0xe800;
 
@@ -146,8 +146,8 @@ smbios_init_type_1(void *start)
     p->header.length = sizeof(struct smbios_type_1);
     p->header.handle = 0x100;
 
-    load_str_field_with_default(1, manufacturer_str, CONFIG_APPNAME);
-    load_str_field_with_default(1, product_name_str, CONFIG_APPNAME);
+    load_str_field_with_default(1, manufacturer_str, CONFIG_SMBIOS_MANUFACTURER);
+    load_str_field_with_default(1, product_name_str, CONFIG_SMBIOS_PROD_NAME);
     load_str_field_or_skip(1, version_str);
     load_str_field_or_skip(1, serial_number_str);
 
@@ -184,7 +184,7 @@ smbios_init_type_3(void *start)
     p->header.length = sizeof(struct smbios_type_3);
     p->header.handle = 0x300;
 
-    load_str_field_with_default(3, manufacturer_str, CONFIG_APPNAME);
+    load_str_field_with_default(3, manufacturer_str, CONFIG_SMBIOS_TYPE3_NAME);
     set_field_with_default(3, type, 0x01); /* other */
 
     load_str_field_or_skip(3, version_str);
@@ -240,7 +240,7 @@ smbios_init_type_4(void *start, unsigned int cpu_number)
     set_field_with_default(4, processor_type, 0x03); /* CPU */
     set_field_with_default(4, processor_family, 0x01); /* other */
 
-    load_str_field_with_default(4, processor_manufacturer_str, CONFIG_APPNAME);
+    load_str_field_with_default(4, processor_manufacturer_str, CONFIG_SMBIOS_CPU_NAME);
 
     if (!qemu_cfg_smbios_load_field(4, offsetof(struct smbios_type_4,
                                     processor_id), p->processor_id)) {
-- 
1.7.3.2

